packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -426,8 +471,22 @@ class _SnackBarState extends State<SnackBar> {
     );
 
     if (isFloatingSnackBar) {
-      snackBar = Padding(
-        padding: const EdgeInsets.fromLTRB(15.0, 5.0, 15.0, 10.0),
+      // If width is provided, do not include horizontal margins.
+      if (widget.width != null) {
+        snackBar = Container(
+          margin: const EdgeInsets.only(top: 5.0, bottom: 10.0),
+          width: widget.width,
+          child: snackBar,
+        );
+      } else {
+        snackBar = Padding(
+          padding: widget.margin ?? const EdgeInsets.fromLTRB(15.0, 5.0, 15.0, 10.0),
```

Comment:
nit: top and bottom padding could be constants

Reply:
done